File: Engine/Source/Components/MeshRenderer.cpp
#include <Yonai/Scripting/Assembly.hpp>
#include <Yonai/Components/Component.hpp>
#include <Yonai/Scripting/InternalCalls.hpp>
#include <Yonai/Components/MeshRenderer.hpp>

using namespace Yonai;
using namespace Yonai::Scripting;
using namespace Yonai::Components;

ADD_MANAGED_GET_SET(MeshRenderer, Mesh, uint64_t)
ADD_MANAGED_GET_SET(MeshRenderer, Material, uint64_t)

File: Apps/YonaiScriptCore/Components/MeshRenderer.cs
using Yonai.Graphics;

namespace Yonai
{
	public class MeshRenderer : Component
	{
		[Serialize(false)]
		public UUID MeshID { get; set; }

		[Serialize(false)]
		public UUID MaterialID { get; set; }

		public Mesh Mesh
		{
			get => Resource.Get<Mesh>(MeshID);
			set => MeshID = value?.ResourceID ?? UUID.Invalid;
		}

		public Material Material
		{
			get => Resource.Get<Material>(MaterialID);
			set => MaterialID = value?.ResourceID ?? UUID.Invalid;
		}
	}
}

